arrays/arrays.c

```diff
@@ -114,33 +149,63 @@ void arr_append(Array *arr, char *element) {
  *
  * Throw an error if the value is not found.
  *****/
-void arr_remove(Array *arr, char *element) {
 
+// adding an additional function to search
+// the array for the index of an element
+// to both use in arr_remove and to have
+// accessable in general
+int arr_find_index(Array *arr, char *element)
+{
+  for (int i = 0; i < arr->count; i++)
+  {
+    if (strcmp(arr->elements[i], element) == 0)
+    {
+      return i;
+    }
+  }
+  return -1;
+}
+
+void arr_remove(Array *arr, char *element)
+{
   // Search for the first occurence of the element and remove it.
   // Don't forget to free its memory!
+  int index = arr_find_index(arr, element);
+  if (index == -1)
+  {
+    printf("The requested element was not found in the array");
+    return;
+  }
+  free(arr->elements[index]);
 
   // Shift over every element after the removed element to the left one position
+  for (int i = index; i < arr->count; i++)
+  {
+    arr->elements[i] = arr->elements[i + 1];
+  }
+  arr->elements[arr->count] = '\0';
```

Comment:
There's no need to add a null terminator at the end of your array. Also this is an array of pointers, not of characters, so if you were going to do this, rather than use a null terminator character, you want to use it's pointer equivalent, the null pointer A.K.A `NULL`